examples/deal.II/bp.cc

```diff
@@ -0,0 +1,235 @@
+          }
+        else
+          {
+            AssertThrow(false, ExcNotImplemented());
+          }
+
+
+        argc -= 2;
+        argv += 2;
+      }
+
+    return false;
+  }
+};
+
+
+
+int
+main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+  Parameters params;
+  if (params.parse(argc - 1, argv + 1))
+    return 0;
+
+  ConditionalOStream pout(std::cout, Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0);
+
+  //  configuration
+  const BPType bp = params.bp;
+
+  using Number                     = double;
+  using VectorType                 = LinearAlgebra::distributed::Vector<Number>;
+  const unsigned int dim           = 2;
+  const unsigned int fe_degree     = params.fe_degree;
+  const unsigned int n_q_points    = (bp <= BPType::BP4) ? (fe_degree + 2) : (fe_degree + 1);
+  const unsigned int n_refinements = params.n_global_refinements;
+  const unsigned int n_components =
+    (bp == BPType::BP1 || bp == BPType::BP3 || bp == BPType::BP5) ? 1 : dim;
+
+  // create mapping, quadrature, fe, mesh, ...
+  MappingQ1<dim> mapping;
+  QGauss<dim>    quadrature(n_q_points);
+  FESystem<dim>  fe(FE_Q<dim>(fe_degree), n_components);
+
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+  GridGenerator::hyper_cube(tria);
+  tria.refine_global(n_refinements);
+
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+
+  AffineConstraints<Number> constraints;
+
+  if (!(bp == BPType::BP1 || bp == BPType::BP2))
+    {
+      // for stiffness matrix
+      DoFTools::make_zero_boundary_constraints(dof_handler, constraints);
+      constraints.close();
+    }
+
+  DoFRenumbering::support_point_wise(dof_handler);
+
+  const auto test = [&](const std::string &label, const auto &op) {
+    (void)label;
+
+    // initialize vector
+    VectorType u, v;
+    op.initialize_dof_vector(u);
+    op.initialize_dof_vector(v);
+    u = 1.0;
+
+    constraints.set_zero(u);
+
+    // perform matrix-vector product
+    op.vmult(v, u);
+
+    // create solver
+    ReductionControl reduction_control(100, 1e-20, 1e-6);
+
+    // create preconditioner
+    DiagonalMatrix<VectorType> diagonal_matrix;
+    op.compute_inverse_diagonal(diagonal_matrix.get_vector());
+
+    std::chrono::time_point<std::chrono::system_clock> now;
+
+    try
+      {
+        // solve problem
+        SolverCG<VectorType> solver(reduction_control);
+        now = std::chrono::system_clock::now();
+        solver.solve(op, v, u, diagonal_matrix);
+      }
+    catch (const SolverControl::NoConvergence &)
+      {
+        // nothing to do
+      }
+
+
+    const auto time =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now() - now)
+        .count() /
+      1e9;
+
+
+    pout << label << ": " << reduction_control.last_step() << " " << v.l2_norm() << " "
+         << (params.print_timings ? time : 0.0) << std::endl;
+  };
```

Comment:
```suggestion
    if (params.print_timings || SOME SOLUTION ERROR TOLERANCE)
      {
          pout << label << ": " << reduction_control.last_step() << " " << v.l2_norm() << " "
             << time << std::endl;
       }
  };
```
I think something like this would give us no output during testing. Our testing script treats all output as an indication of an error.

Reply:
OK. I have made the change. I am surprised that you don't check the iteration numbers in CI.

Reply:
We usually use the L2 error between the target and computed solution to determine correctness in situations like this where there is an analytic solution